### PR TITLE
Format Library: Add alt edit field to inline image

### LINKED ISSUE #63323: No ability to edit Inline Image alternative text in context
### Description

Currently, to the best of my knowledge, it is only possible edit an inline image's global alt text (without editing HTML). This is an issue as most alt text should be contextual. I'm currently in a place where I need my inline images to have empty alt text while it's very important that they retain alt text for use in galleries throughout the site.

### Step-by-step reproduction instructions

1. Insert inline image
2. You can't edit the alt text

### Screenshots, screen recording, code snippet

_No response_

### Environment info

- WP 6.6-RC2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes #63323
Related to #28813

## What?

This PR makes it possible to change the alt text in the inline image popover.

image

## Why?

If the media already has alt text set, the alt attribute will be applied to the inserted inline image. However, if you want to change the alt attribute, you have no choice but to change the alt of the original media and then reinsert it, or change the value directly via the code editor.

## How?

Add a text area to the popover to change the alt text.

## Testing Instructions

- Add an inline image.
- Select the image.
- Change alt text and click the Apply button.
- If you check the HTML, you'll see that the alt text has changed.

### Testing Instructions for Keyboard

You should be able to change the alt text using just the keyboard.

## Screenshots or screencast 

https://github.com/user-attachments/assets/73a495ad-91a4-4bae-a49e-a1efb3c2bc97

### CODE REVIEW

**Review by jasmussen — APPROVED** [review]
This could use some style improvements. But whether they happen here or as a followup, that can be decided separately. 

Mainly, the input sizes, spacings, could use some tweaks. The modal shows a 12px gap between last input, and the confirm button:
Screenshot 2024-07-31 at 09 03 32

I think that should actually be 16, but that's separate. But that same gap should also be the gap between the width control, and the alt text control, right now they're very close together:

Screenshot 2024-07-31 at 09 03 56

Question: should the same help text be present for this alt-text field?
Screenshot 2024-07-31 at 09 03 47



Nice work.

File: packages/format-library/src/image/index.js [hunk 1]
```diff
@@ -73,24 +77,43 @@ function InlineUI( { value, onChange, activeObjectAttributes, contentRef } ) {
...
-						label={ __( 'Apply' ) }
-						type="submit"
-						__next40pxDefaultSize
+					<TextareaControl
+						label={ __( 'Alternative text' ) }
+						__nextHasNoMarginBottom
+						value={ alt }
+						onChange={ ( newAlt ) => {
+							setAlt( newAlt );
+							setHasChanged( true );
```
Comment (ellatrix) [inline packages/format-library/src/image/index.js]: Why is this needed? Can't we compare the current state?

Comment (t-hamano) [reply packages/format-library/src/image/index.js]: That's true. I removed the redundant state by cf66b8e. The approach is similar to [this one](https://github.com/WordPress/gutenberg/blob/d5e7a77eb851329da16510f758bdd742099fb5ac/packages/block-editor/src/components/block-rename/modal.js#L29-L31).

If there is a change from the initial value, the Apply button should become active:

https://github.com/user-attachments/assets/b3c695b0-4334-421a-a91b-5b67a72cbf32

**Comment by t-hamano** [comment]
> Question: should the same help text be present for this alt-text field?

I was worried that the popover would be too big, but I gave it a try. What do you think?

image

---

Update:

> I think that should actually be 16, but that's separate. But that same gap should also be the gap between the width control, and the alt text control, right now they're very close together:

Changed the gap to 16px.

image

**Comment by ellatrix** [comment]
If it becomes too big, maybe there should be a tab in the inspector for inline content. I think @dmsnell was also looking for that for Bits. :)

File: packages/format-library/src/image/index.js [hunk 2]
```diff
@@ -76,21 +82,54 @@ function InlineUI( { value, onChange, activeObjectAttributes, contentRef } ) {
...
+							disabled={ ! hasChanged }
+							accessibleWhenDisabled
+							variant="primary"
+							type="submit"
+							size="compact"
+						>
+							{ __( 'Apply' ) }
+						</Button>
+					</HStack>
+				</VStack>
```
Comment (ellatrix) [inline packages/format-library/src/image/index.js:132]: Looking at this deeper, why do we even have an apply button here? The image block does not have this either, it just updates right away.

**Review by ellatrix — APPROVED** [review]

File: packages/format-library/src/image/index.js (hunk 2 above)
Comment (t-hamano) [reply packages/format-library/src/image/index.js:132]: Even in trunk, it won't update unless you explicitly submit the form or press the apply icon button. If we update it immediately, I think the popover will rattle 😅

https://github.com/user-attachments/assets/6085c744-7ba5-46d2-9387-c272b1bd568a

**Comment by t-hamano** [comment]
@jasmussen Please let us know if there is anything we can improve further in terms of design in this PR 🙏 Below are the latest screenshots:

image

**Comment by richtabor** [comment]
### Here's what I'm seeing: 

CleanShot 2024-07-31 at 10 25 24

### I pushed two small changes to ensure this: 

CleanShot 2024-07-31 at 10 29 02

**Review by richtabor — APPROVED** [review]
LGTM 

CleanShot 2024-07-31 at 10 29 02

**Comment by t-hamano** [comment]
@richtabor Thanks for the update!
